newIDE/app/src/MainFrame/index.js

```diff
@@ -1242,13 +1244,19 @@ const MainFrame = (props: Props) => {
 
     if (storageProvider) {
       getStorageProviderOperations(storageProvider).then(() => {
-        openFromFileMetadata(fileMetadata).then(state => {
-          if (state)
-            openSceneOrProjectManager({
-              currentProject: state.currentProject,
-              editorTabs: state.editorTabs,
-            });
-        });
+        openFromFileMetadata(fileMetadata)
+          .then(state => {
+            if (state)
+              openSceneOrProjectManager({
+                currentProject: state.currentProject,
+                editorTabs: state.editorTabs,
+              });
+          })
+          .catch(error => {
+            preferences.removeRecentProjectFile(
+              fileMetadataAndStorageProviderName
+            );
+          });
```

Comment:
If the `fileMetadata` is indeed a sad path, it's best to remove it.
1. The app would try to auto-open a sad path only once.
2. There's no point in storing useless data ;)

Reply:
Sounds good to me!